File: whsk2/lua/corelibs/Level.lua
Level = {}

function Level._updateCall()

	Level.Update()
	local actors = Actor._actors
	for _,actor in pairs(actors) do
		actor:_update()
	end

end

function Level._unloadCall()

	Level.OnUnload()

	Level.OnLoad = function()end
	Level.OnUnload = function()end
	Level.Update = function()end
	Level.OnQuit = function()end

end

return Level

File: whsk2/Level.cpp
#include "Level.h"
#include "Input.h"
#include "EngineManager.h"
#include "LuaLevel.h"

using Engine::Internal::EngineManager;
using Engine::Level;

Level* Level::loadedLevel = nullptr;

void Level::UpdateCall()
{
	if (loadedLevel)
		loadedLevel->Update();
}

void Level::QuitCall(int exitCode)
{
	if (loadedLevel)
		loadedLevel->OnQuit(exitCode);
}

void Level::ResetEnvironment()
{
	Input::Update(0, 0, 0);
	EngineManager::cameraPos = Vector2D(0, 0);
}

void Level::LoadLevel(std::string file)
{
	if (loadedLevel)
	{
		loadedLevel->OnUnload();
		delete loadedLevel;
	}

	LuaLevel* level = new LuaLevel(file);

	loadedLevel = (Level*)(level);
	ResetEnvironment();
	loadedLevel->OnLoad();
}